app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,71 @@
+require 'pry'
+class RentalsController < ApplicationController
+
+  def checkout
+    movie_id = params[:movie_id]
+    customer_id = params[:customer_id]
+
+    date = Date.today
+    rental_params = {
+      checkout: nil,
+      due_date: nil,
+      customer_id: customer_id,
+      movie_id: movie_id
+    }
+
+    movie = Movie.find_by(id: movie_id)
+    customer = Customer.find_by(id: customer_id)
+
+    if movie[:available_inventory] == 0
+      render json: {
+        errors: {
+          available_inventory: ["Movie is currently all checked out. Sorry."]
+        }
+      } , status: :bad_request
+      return
+    end
+
+    new_rental = Rental.new(rental_params)
+    new_rental[:checkout] = date
+    new_rental[:due_date] = date + 7
```

Comment:
You should take all of the logic in this controller action and encapsulate it in a model method. Perhaps something like `Movie#checkout(customer_id)`. You could have it raise an exception if there's not enough inventory available, or implement a custom validation on the rental.